### ISSUE TITLE
[Bug]: Ensure NumPy-compatibility of `test_statistics.py`

### ISSUE THREAD

**Issue by mrfh92** [helmholtz-analytics/heat#1232]
### What happened?

In `test_statistics.py`, NumPy throws the following deprecation warning: 
```
DeprecationWarning: the `interpolation=` argument to percentile was renamed to `method=`, which has additional options.
Users of the modes 'nearest', 'lower', 'higher', or 'midpoint' are encouraged to review the method they used. (Deprecated NumPy 1.22)
```

Although we only use NumPy here to get the "correct" result to which our Heat result is compared, we should keep this up-to-date. 

### Version

1.3.x

### Python version

None

### PyTorch version

None

**Comment by RengarajR404** [comment]
Heyy 
Can you assign me to this issue?

**Comment by mrfh92** [comment]
Further deprecation warning (might depend on NumPy-version used, of course): 
```
test_arithmetics.py:620: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
```

**Comment by mrfh92** [comment]
@RengarajR404 feel free to ask if there are any questions :)

**Comment by samadpls** [comment]
Hey @RengarajR404, if you're not actively working on this issue, can I take it over? Thanks!

**Comment by samadpls** [comment]
Further deprecation warning related to NumPy 1.25:
```
factories.py:107: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation.
```

**Comment by samadpls** [comment]
Hi @mrfh92, could you please assign this task to me?

**Comment by mrfh92** [comment]
has been addressed in #1315 and #1337
